src/economy/economy.cc

```diff
@@ -162,11 +162,11 @@ void Economy::check_splits() {
 		Flag* f2 = split_checks_.back().second.get(egbase);
 		split_checks_.pop_back();
 
-		if (!f1 || !f2) {
-			if (!f1 && !f2) {
+		if ((f1 == nullptr) || (f2 == nullptr)) {
+			if ((f1 == nullptr) && (f2 == nullptr)) {
```

Comment:
The first `if` checks if at least one of `f1` and `f2` is `null`, and the second `if` checks if they are both `null`. So if both variables are `null` we `continue` immediately, if exactly one is `null` (the XOR case) we proceed with the if clause, and if neither is `null` we proceed below. Rewriting this with an XOR wouldn't gain anything IMHO since we'd still need to handle all three cases separately.

Reply:
well I thought continuing would be the same as not perfoming the if loop (jumping directly to line 188). so we execute the statements in this if (from 188 to 187) only if one of them is nullptr which is an XOR.
However it doesn't matter it was just a drive by finding.

> The first `if` checks if at least one of `f1` and `f2` is `null`, and the second `if` checks if they are both `null`. So if both variables are `null` we `continue` immediately, if exactly one is `null` (the XOR case) we proceed with the if clause, and if neither is `null` we proceed below. Rewriting this with an XOR wouldn't gain anything IMHO since we'd still need to handle all three cases separately.

Reply:
The `continue` keyword only operates on loops: `for`, `while`, and `do while`. For all other purposes its equivalent would be a `goto` statement, which is avoided wherever possible by nesting `if`s instead.

Reply:
ok thanks for clarification